### DISCUSSION TITLE
Limit CD-ROM speeds to common values

### DISCUSSION THREAD

**Discussion by chungy** [86Box/86Box#3341] [Ideas]
**Is your feature request related to a problem? Please describe.**

The current combobox for CD-ROM speeds simply enumerates all values from 1× to 72×, fills the entire screen, and it's mostly a lot of clutter especially for the nearly-nonexistent odd-number speeds (besides 1× and one example of a 3× that was posted on Discord).

It would vastly simplify the user interface and experience to limit the speed selections to a few desirable values, and ones that existed in real hardware.

**Describe the solution you'd like**

A list that only has a few values, a proposed list of values from myself: 1, 2, 3, 4, 8, 12, 16, 24, 32, 48, 52, 72

**Comment by cold-brewed** [comment]
This seems reasonable to me. Even with this change a user would still be able to edit the parameter in the config file to manually select a different speed.

**Comment by OBattler** [comment]
6x drives exist, we bought a Goldstar one for our 486 back in the day.

**Comment by waltje** [comment]
See the table in my code.  And yes, 6x existed, and were fairly common then.


    On Monday, February 20, 2023 at 01:40:44 PM GMT-5, Miran Grča ***@***.***> wrote:  
 
 


6x drives exits, we bought a Goldstar one for our 486 back in the day.

**Comment by TheVirtualComputer** [comment]
36x drives also exist, as well as 6.7x drives. Also, some people may prefer to customize their CD speeds to their own preference.

**Comment by chungy** [comment]
The spirit of the feature request was not to replicate every speed that has ever existed, but limit the values to some common and useful ones. Declutter the UI :)

From discussion on the 86box chat, there is another possibility that speed might be tied to drive models (as HEAD allows for many different ones) instead of a selectable option.

**Comment by TheVirtualComputer** [comment]
But you would then have to add many different CD-ROM and/or DVD-ROM drives in order to accommodate for the differences in drive speed, which would probably be a major task in itself. And even then, you would also have to account for the fact that there were also SCSI drives for each set of CD-ROM speeds in addition to IDE/EIDE/ATAPI drives, which would further add to the amount of work that would be required.

Also, what if you wanted to emulate a particular model of PC that shipped with a CD-ROM drive that ran at a certain speed, but the emulator didn't allow it? You would then have to download the source for the emulator and add the drive manually (assuming that the ability to change CD-ROM drive speeds manually is also removed from the configuration files).

**Comment by FavoritoHJS** [comment]
Maybe change the dropdown to a number input slider or selector? So if you want a 32x drive, type 32x instead of searching for the option that says 32x
